terraform/main.tf

```diff
@@ -0,0 +1,290 @@
+  has_downloads               = each.value.has_downloads
+  has_discussions             = each.value.has_discussions
+  has_issues                  = true
+  has_projects                = true
+  has_wiki                    = false
+  is_template                 = each.value.is_template
+  name                        = each.key
+  squash_merge_commit_message = "BLANK"
+  squash_merge_commit_title   = "PR_TITLE"
+  topics                      = each.value.topics
+  visibility                  = each.value.visibility
+  vulnerability_alerts        = false
+
+  dynamic "template" {
+    for_each = each.value.template != null ? [each.value.template] : []
+
+    content {
+      owner      = "django-commons"
+      repository = template.value
+    }
+  }
+}
+
+# GitHub Repository File Resource
+# https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_file
+#
+# resource "github_repository_file" "security_policy" {
+#   for_each = var.repositories
+#
+#   branch              = "main"
+#   content = templatefile("${path.module}/markdown/SECURITY.md.tpl", { repository = each.key })
+#   file                = "SECURITY.md"
+#   repository          = each.key
+#   commit_message      = "Update SECURITY.md"
+#   commit_author       = "Open Source Infrastructure as Code Service Account"
+#   commit_email        = "github-sa@osinfra.io"
+#   overwrite_on_create = true
+#
+#   depends_on = [
+#     github_branch_protection.this
+#   ]
+# }
+
+
+# Github Team Resource
+# https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team
+
+# If you need to import a team, you can do so with the following command:
+# terraform import github_team.this\[\"google-cloud-platform\"\] <team_id>
+
+# To get the team ids, you can run the following curl command with a token that has the read:org scope against your own organization.
+# curl -H "Authorization: token $GITHUB_READ_ORG_TOKEN" https://api.github.com/orgs/osinfra-io/teams
+
+resource "github_team" "parents" {
+  for_each = var.team_parents
+
+  name        = each.key
+  description = each.value.description
+  privacy     = each.value.privacy
+}
+
+resource "github_team" "children" {
+  for_each = var.team_children
+
+  description    = each.value.description
+  name           = each.key
+  parent_team_id = github_team.parents[each.value.parent_team_key].id
+  privacy        = github_team.parents[each.value.parent_team_key].privacy
+}
+
+# GitHub Team Membership Resource
+# https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team_members
+
+resource "github_team_members" "parents" {
+  for_each = var.team_parents
+
+  team_id = github_team.parents[each.key].id
+
+  dynamic "members" {
+    for_each = each.value.members
+
+    content {
+      username = members.value
+      role     = "member"
+    }
+  }
+
+  dynamic "members" {
+    for_each = each.value.maintainers
+
+    content {
+      username = members.value
```

Comment:
Is this supposed to be members or maintainers?

Reply:
Good catch. I assume this also leftovers from the boilerplate code I used.
This is not used anywhere, so I will remove it.

Reply:
Actually, this is under `resource "github_team_members" "parents"`
So notice a team is created with two sections of `members`, one for `members` and another for `maintainers`.

```

resource "github_team_members" "parents" {
  for_each = var.team_parents

  team_id = github_team.parents[each.key].id

  dynamic "members" { # <<< THIS IS WHERE WE CREATE REGULAR TEAM MEMBERS
    for_each = each.value.members

    content {
      username = members.value
      role     = "member"
    }
  }

  dynamic "members" {  # <<< THIS IS WHERE WE CREATE MAINTAINERS TEAM MEMBERS
    for_each = each.value.maintainers

    content {
      username = members.value
      role     = "maintainer"
    }
  }
}
```

Reply:
I'm confused by `username = members.value` in the maintainers part. Is it fetching it based on the `dynamic "members"` part? If so, maybe we should rename `each.value.members` to something else so that's easier to understand?

Reply:
Let's say:
```
maintainers = [
      "tim-schilling",
      "williln",
      "ryancheley",
    ]
```
So this:
```
dynamic "members" {  # <<< THIS IS WHERE WE CREATE MAINTAINERS TEAM MEMBERS
    for_each = each.value.maintainers

    content {
      username = members.value
      role     = "maintainer"
    }
  }
```
Will result in:
```
"members" {
      username = "tim-schilling"
      role     = "maintainer"
}
"members" {
      username = "williln"
      role     = "maintainer"
}
"members" {
      username = "ryancheley"
      role     = "maintainer"
}

```